--- gobject-introspection-1.44.0/giscanner/giscannermodule.c.orig	2014-08-04 10:37:07 -0400
+++ gobject-introspection-1.44.0/giscanner/giscannermodule.c	2015-07-19 19:40:48 -0400
@@ -427,7 +427,7 @@
   if (!PyArg_ParseTuple (args, "i:SourceScanner.parse_file", &fd))
     return NULL;
 
-#ifdef _WIN32
+#if 0
   /* The file descriptor passed to us is from the C library Python
    * uses. That is msvcr71.dll for Python 2.5 and msvcr90.dll for
    * Python 2.6, 2.7, 3.2 etc; and msvcr100.dll for Python 3.3 and later.
--- gobject-introspection-1.44.0/giscanner/gdumpparser.py.orig	2015-07-19 19:38:24 -0400
+++ gobject-introspection-1.44.0/giscanner/gdumpparser.py	2015-07-19 19:42:09 -0400
@@ -499,6 +499,10 @@
         except TransformerException as e:
             message.fatal(e)
         pair_node = self._namespace.get(name)
+        while isinstance(pair_node, ast.Alias):
+            pair_node = pair_node.target
+            assert pair_node
+            pair_node = self._namespace.get(pair_node.unresolved_string)
         if not pair_node:
             # Keep the "bare" boxed instance
             self._namespace.append(boxed)
--- gobject-introspection-1.44.0/giscanner/dumper.py.orig	2014-08-05 10:34:40 -0400
+++ gobject-introspection-1.44.0/giscanner/dumper.py	2015-07-19 20:11:05 -0400
@@ -104,7 +104,10 @@
         # We have to use the current directory to work around Unix
         # sysadmins who mount /tmp noexec
         tmpdir = tempfile.mkdtemp('', 'tmp-introspect', dir=os.getcwd())
-        os.mkdir(os.path.join(tmpdir, '.libs'))
+        dot_libs_dir = os.path.join(tmpdir, '.libs')
+
+        if not os.path.isdir(dot_libs_dir):
+            os.mkdir(dot_libs_dir)
 
         tpl_args = {}
         if self._uninst_srcdir is not None:
@@ -246,7 +249,7 @@
             raise CompilerError(e)
 
     def _link(self, output, *sources):
-        args = []
+        args = ['C:/msys/bin/bash.exe']
         libtool = utils.get_libtool_command(self._options)
         if libtool:
             args.extend(libtool)
