--- libxml2-2.9.2/configure.ac.orig	2015-01-08 16:31:45 -0500
+++ libxml2-2.9.2/configure.ac	2015-01-08 16:32:45 -0500
@@ -68,7 +68,7 @@
 AC_PATH_PROG(WGET, wget, /usr/bin/wget)
 AC_PATH_PROG(XMLLINT, xmllint, /usr/bin/xmllint)
 AC_PATH_PROG(XSLTPROC, xsltproc, /usr/bin/xsltproc)
-PKG_PROG_PKG_CONFIG
+# PKG_PROG_PKG_CONFIG
 
 LT_INIT
 
@@ -424,14 +424,14 @@
 else
     # Try pkg-config first so that static linking works.
     # If this succeeeds, we ignore the WITH_LZMA directory.
-    PKG_CHECK_MODULES([LZMA],[liblzma],
-        [have_liblzma=yes],
-        [have_liblzma=no])
+    # PKG_CHECK_MODULES([LZMA],[liblzma],
+    #     [have_liblzma=yes],
+    #     [have_liblzma=no])
 
      # If pkg-config failed, fall back to AC_CHECK_LIB. This
      # will not pick up the necessary LIBS flags for liblzma's
      # private dependencies, though, so static linking may fail.
-     if test "x$have_liblzma" = "xno"; then
+     # if test "x$have_liblzma" = "xno"; then
          AC_CHECK_HEADERS(lzma.h,
 	    [SAVE_LDFLAGS="${LDFLAGS}"
 	     LDFLAGS="-L${LZMA_DIR}/lib"
@@ -445,7 +445,7 @@
                 fi],
                 [have_liblzma=no])
 	     LDFLAGS="${SAVE_LDFLAGS}"])
-    fi
+    # fi
 
     # Found the library via either method?
     if test "x$have_liblzma" = "xyes"; then
